src/internal/operators/find.ts

```diff
@@ -51,28 +50,31 @@ export function find<T>(predicate: (value: T, index: number, source: Observable<
  * matches the condition.
  */
 export function find<T>(predicate: (value: T, index: number, source: Observable<T>) => boolean): OperatorFunction<T, T | undefined> {
-  return operate(createFind(predicate, 'value'));
+  return (source) => new Observable((subscriber) => createFind(predicate, 'value', source, subscriber));
 }
 
-export function createFind<T>(predicate: (value: T, index: number, source: Observable<T>) => boolean, emit: 'value' | 'index') {
+export function createFind<T>(
+  predicate: (value: T, index: number, source: Observable<T>) => boolean,
+  emit: 'value' | 'index',
+  source: Observable<T>,
+  subscriber: Subscriber<any>
```

Comment:
Higher order function no longer needed here. Affects `reduce` and `scan` a little.